Overhaul of documentation

jlews opened the pull request:
- Documented missing pieces.
- Documented new `useTools` hook.
- Added contents menu
- Split docs into separate files.

patrickbenjamin1 commented on docs/config.md:
````diff
@@ -0,0 +1,84 @@
+
+```TypeScript
+import { ApiHooks } from "@rocketmakers/api-hooks"
+import { apiClient } from "*API CLient location*"
````
typo with CLient rather than Client

could also make it clearer that it's meant to be a path to a project document if written like `/path/to/my/apiclient` which I've seen before

patrickbenjamin1 commented on docs/config.md:
````diff
@@ -0,0 +1,84 @@
+```TypeScript
+import { apiHooks } from "*create method location*"
+
+const MyComponent: React.FunctionComponent = () => {
````
React.FunctionComponent 😎

patrickbenjamin1 commented on docs/config.md:
```diff
@@ -0,0 +1,84 @@
+
+  const [{data, isFetching}] = apiHooks.firstController.getEndpoint.useQuery({
+    caching: {
+      staleIfOlderThan: 10000
```
Worth using your little time utils for this or nah?

judithrocketmakers replied:
Not in an example, surely?

jlews replied:
I've now added an explanation of both approaches in the new "caching section"

patrickbenjamin1 commented on docs/hooks.md:
```diff
@@ -0,0 +1,259 @@
+
+const MyComponent: React.FunctionComponent = () => {
+
+  const [{data, isFetching}] = apiHooks.firstController.getEndpoint.useQuery();
```
Very very minor and unimportant thing, but the formatting here is slightly different to elsewhere:

```ts
[{data, isFetching}]
```

vs

```ts
[{ data, isFetching }]
```

patrickbenjamin1 commented on docs/hooks.md:
````diff
@@ -0,0 +1,259 @@
+```TypeScript
+import { apiHooks } from "*create method location*"
+
+const MyComponent: React.FunctionComponent<{ userId: string }> = (userId) => {
````
should be:

```tsx
const MyComponent: React.FunctionComponent<{ userId: string }> = ({ userId }) => {
```

the prop isn't being destructured

patrickbenjamin1 commented on docs/quirks.md:
```diff
@@ -0,0 +1,55 @@
+
+const MyComponent: React.FunctionComponent<{ userId?: string }> = (props) => {
+
+  const [{data, isFetching}, getUserFetch] = apiHooks.users.getUser.useQuery({
```
formatting as above, for consistency

```
  const [{ data, isFetching }, getUserFetch] = apiHooks.users.getUser.useQuery({
```

judithrocketmakers commented on docs/advanced.md:
```diff
@@ -0,0 +1,292 @@
+#### For example:
+
+- Our application has an infinitely paged list of users returned as an array of user objects from a `getUsers` query.
+- We also have a `getUser` query which receives an identical user object, but for a single `id`, it uses this `id` as it's cache key.
```
its

judithrocketmakers commented on docs/hooks.md:
```diff
@@ -0,0 +1,259 @@
+NOTE:
+
+- By default, queries will be invoked when the component loads, so if you really want to send the request manually, you'll need to remember the `autoInvoke: false` setting.
+- Manual fetches will always attempt to fetch from the server, regardless of any valid cache, this can be overridden by passing `{forceNetwork: false}` to the second argument of the manual fetch function.
```
Spacing on `{forceNetwork: false}` as per other object-based comments?

judithrocketmakers commented on docs/hooks.md:
```diff
@@ -0,0 +1,259 @@
+
+NOTE:
+
+- It's important to note that queries can _not_ be used in this was, as for the caching system to function properly, queries can't return a promise like mutations can.
```
'in this way'?

judithrocketmakers commented on docs/hooks.md:
```diff
@@ -0,0 +1,259 @@
+  const [userList, setUserList] = React.useState([])
+
+  const populateUserListFromInput = React.useCallback((searchInput) => {
+    const newUserList = await searchUsers({query: searchInput})
```
Object spacing?

jlews commented:
Thanks @patrickbenjamin1 and @judithrocketmakers for your reviews. I've resolved all of your comments.

I've also refactored things a bit and added a new "caching" section this morning. I felt like the caching logic and reasons behind it weren't adequately explained. If you have the time to take a quick look at `caching.md` this morning, that would be fantastic!

I promise this is the final version!
